src/python/pants/backend/jvm/subsystems/jar_dependency_management.py

```diff
@@ -361,19 +362,24 @@ def _compute_artifact_sets(self, targets):
   def _library_targets(self, managed_jar_dependencies):
     for spec in managed_jar_dependencies.library_specs:
```

Comment:
This should probably be flattened to:
```
targets = [t for spec in managed_jar_dependencies.library_specs for t in self.context.resolve(spec)]
for t in Target.closure_for_targets(targets):
  yield t
```

Each closure call does a graph walk, so it's important to merge them where possible.